src/mds/CDir.cc

```diff
@@ -2031,14 +2032,14 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
 	   << ": " << err.what() << dendl;
       clog->warn() << "Corrupt fnode header in " << dirfrag() << ": "
 		   << err.what() << " (" << get_path() << ")";
-      go_bad(complete);
+      go_bad(true);
       return;
     }
     if (!p.end()) {
       clog->warn() << "header buffer of dir " << dirfrag() << " has "
 		  << hdrbl.length() - p.get_off() << " extra bytes ("
                   << get_path() << ")";
-      go_bad(complete);
+      go_bad(true);
```

Comment:
Is there any place that can call go_bad() with the argument not equal to true?

Reply:
Not at the moment. But I'd prefer to keep the parameter till getting more feedback.

Reply:
scrubbing is mostly a read-only op ... unless *repair* is specified
however, there's not much that can be repaired via scrubbing if a dirfrag object goes missing on disk
maybe we could retry fetching the OMAP a specified number of times in the scrub  path after which we could flag an error rather than loop indefinitely ... while also keeping the OMAP fetch path untouched

Reply:
Looks like doing this will make scrub progress rather than terminating it, yes? cc @mchangir @ifed01